examples/chef/common/clusters/occupancy-sensing/occupancy-sensing-stub.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/CommandHandler.h>
+#include <app/clusters/occupancy-sensor-server/occupancy-hal.h>
+#include <app/clusters/occupancy-sensor-server/occupancy-sensor-server.h>
+#include <app/util/attribute-storage.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#ifdef MATTER_DM_PLUGIN_OCCUPANCY_SENSING_SERVER
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OccupancySensing;
+using namespace chip::app::Clusters::OccupancySensing::Structs;
+using namespace chip::DeviceLayer;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace {
+
+static constexpr size_t kOccupancySensingClusterTableSize =
+    MATTER_DM_OCCUPANCY_SENSING_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+static_assert(kOccupancySensingClusterTableSize <= kEmberInvalidEndpointIndex, "Occupancy Sensing Cluster table size error");
+
+static std::unique_ptr<Instance> gOccupancySensingClusterInstances[kOccupancySensingClusterTableSize];
+
+} // namespace
+
+void emberAfOccupancySensingClusterInitCallback(EndpointId endpointId)
```

Comment:
This does not work at all --> it doesn't implement the hold time timer. The hold time is supposed to ensure that the change of state doesn't occur until at least after the amount of hold time. This sample is missing APIs to allow simulating the occupancy state in a way that honors the hold time.

Note that the SDK cluster implementation doesn't actually support the hold time feature by itself and requires product integration support that is not trivial.

This PR cannot land until the TC-OCC-* tests pass on the sample.